src/main/java/codec/Codec.java

```diff
@@ -37,21 +37,21 @@ public ByteBuffer encode(Message message) throws Exception {
 
             return messageBuffer;
         } else {
-            throw new Exception("Max message size = " + MAX_LENGTH + " | Your message size is " + message.getMessage().length());
+            throw new IndexOutOfBoundsException("Max message size = " + MAX_LENGTH + " | Your message size is " + message.getMessage().length());
```

Comment:
`IndexOutOfBoundsException` should be thrown when we deal with collections and somebody tries to access the element with index that is out of bounds. But we do not operate with any indexes here at all. So, another type of exception should be chosen. You can use IlllegalArgumentException e.g.